src/ui/mod.rs

```diff
@@ -78,6 +78,12 @@ impl<R: ResourcesProvider> Ui<R> {
                     service::submission_info(&mut self.res, &scope, submission_id, false)?;
                 submission::print_submission_info(self, &scope, submission_info, false)?;
             }
+            Command::Submission(scope, Submission::NthLatest(task_id, n)) => {
```

Comment:
Consider combining this with the preceding match arm, and having another match to discriminate between `Id` and `NthLatest`, since only the `let submission_info =` part is changed.